packages/core/src/schema.ts

```diff
@@ -105,7 +105,7 @@ export class SchemaValidator extends Validator {
     }
 
     // 3) Validate value
-    return validator.validatePath(value, path, ctx);
+    return validator.validatePath(value, path, ctx) as PromiseLike<ValidationResult>;
```

Comment:
This cast doesn't seem to be necessary?  My editor is resolving it as `PromiseLike<ValidationResult<unknown>>`.